loader/db.go

```diff
@@ -149,23 +149,40 @@ func (conn *Conn) executeSQL(ctx *tcontext.Context, queries []string, args ...[]
 	return err
 }
 
-func createConn(cfg *config.SubTaskConfig) (*Conn, error) {
-	dbDSN := fmt.Sprintf("%s:%s@tcp(%s:%d)/?charset=utf8mb4&maxAllowedPacket=%d",
-		cfg.To.User, cfg.To.Password, cfg.To.Host, cfg.To.Port, *cfg.To.MaxAllowedPacket)
-	baseConn, err := baseconn.NewBaseConn(dbDSN, &retry.FiniteRetryStrategy{})
-	if err != nil {
-		return nil, terror.WithScope(terror.DBErrorAdapt(err, terror.ErrDBDriverError), terror.ScopeDownstream)
+// Close release db connection resource, return it to BaseDB.db connection pool
+func (conn *Conn) Close() error {
+	if conn == nil || conn.baseConn == nil {
+		return nil
 	}
-
-	return &Conn{baseConn: baseConn, cfg: cfg}, nil
+	return conn.baseConn.Close()
 }
 
-func closeConn(conn *Conn) error {
-	if conn.baseConn == nil {
-		return nil
+func createConn(ctx context.Context, cfg *config.SubTaskConfig) (*conn.BaseDB, *Conn, error) {
+	baseDB, err := conn.DefaultDBProvider.Apply(cfg.To, nil)
+	if err != nil {
+		return nil, nil, terror.WithScope(terror.DBErrorAdapt(err, terror.ErrDBDriverError), terror.ScopeDownstream)
```

Comment:
is ScopeUpstream?

Reply:
It's connect to cfg.To DB, so it's kind of ScopeDownstream